packages/editor/src/components/provider/index.js

```diff
@@ -301,15 +314,15 @@ export const ExperimentalEditorProvider = withRegistryProvider(
 
 		// Sets the right rendering mode when loading the editor.
 		useEffect( () => {
-			setRenderingMode( settings.defaultRenderingMode ?? 'post-only' );
-		}, [ settings.defaultRenderingMode, setRenderingMode ] );
+			setRenderingMode( mode );
+		}, [ mode, setRenderingMode ] );
 
 		useHideBlocksFromInserter( post.type, mode );
 
 		// Register the editor commands.
 		useCommands();
 
-		if ( ! isReady ) {
+		if ( ! isReady || ! mode ) {
```

Comment:
One potential here is that the mode can be defined even if the "postType" didn't load yet because of the "post-only" fallback. It seems better to avoid rendering anything until "postType" is defined.

Reply:
Added a check to see if the post type selector has resolved. However I still see an unstyled flash of content 🤔